### PR TITLE
chore: update-api-spec

### PR DESCRIPTION
**Pull request by agraebe**
irt #160 

Outstanding tasks:
- [x] Better documentation for core node endpoints
- [x] Add search endpoint documentation
- [x] Better document request and response fields
- [x] Add address endpoint documentation
- [x] Add "Run in Postman" button (based on new Postman collection)

FYI: I'll frequently update the hosted docs, find them here: https://blockstack.github.io/stacks-blockchain-sidecar/

### CODE REVIEW

File: docs/api/core-node/get-account-data.example.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
...
...365404bd03024301a60b70db8b730e37d967078b922c71a7e0fcae3539665aeff5adc9a101a0110337a5909f5e8d56428e77bd8131ebfa68f2f3bf90bed70e7ea5f5cb23553ac1a517d6ceba5ad141b3a1275322d46475e422b0ec6420270b39d67f987f5c8f7813abbd2203a08f22296421801d473d804b19c4bbcd6132e9100f2ec9aa70b1cd1071b37ce117c0f1191a97b44df912e762f1748ab543e0b40d072293a5a03d51f3bf365e9ce6ffd599422826fcdbd2003871b353002fad638d0f42d36d9db7b39b33ea8e65e4aa01a76734464b908db04e73038040df3d703b65357d0fdbe5bc83907c0500000000000000030000000b2703ee01f3db0cbd3e1ba08405564eed2671533e2509dfc19cc1e1c18fa1fde03868eb120025526d5a06f93353ae65afd8a923c3704fbc603fde398981ab5a55b96f80eaf1221ca0998971764a58269bce1081a3513412213db789632400a9665f2749c93ed0e45184f16b4427948ff4f5c3d50da1aa5567ad850ff18509322676e12142567022b40231ec4e1be7ff397eb842e98a88bc2ece30b25e17c60a372d3e0dc86a3c1c61022db95424b40f156c2f8c5fe0641e981f7c4994c861b22c4ab8fb1adfc2ecca576c7f3a5e43c53e84ea315da76fca30b3e828ff18ee4a491d10d4aecb60a2ddb360e473010e37a6ad59962661b774ea793fb08fb0115cbdef31f3b08bb4d1c350a8b32ff847121f1f4eab06c5298e3047dc648f7d62436a5014562ea1b51d82a31aa6d639f09ae449b4b7a7d55b389bb97b053539b1d3e68ae8ce791ded714350333cd5092df076208dd9a40b965afb01d5c486ff2fb044",
```
Comment (zone117x) [inline docs/api/core-node/get-account-data.example.json:4]: Should we make these long arbitrary hex strings into something like `"balance_proof": "0xabcd"`, for file readability?

**Comment by kyranjamie** [comment]
Thanks for taking this on @agraebe, I'll benefit from these types. 

We also need docs for `/address/:stxAddress/balances` and `/address/:stxAddress/transactions`. Let me know if you need any help. 👍

File: docs/api/core-node/get-account-data.example.json (hunk 1 above)
Comment (agraebe) [reply docs/api/core-node/get-account-data.example.json:4]: We should, I'll make the change :)

File: docs/api/address/get-address-assets.example.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (kyranjamie) [inline docs/api/address/get-address-assets.example.json:1]: This file has 4 space tabs, everywhere else is 2. `prettier --write docs/**/*.json`

Comment (agraebe) [reply docs/api/address/get-address-assets.example.json:1]: thanks, fixed!

**Comment by agraebe** [comment]
@kyranjamie @zone117x this is ready for review

File: docs/api/address/get-address-transactions.schema.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    "offset": {
+      "type": "integer"
+    },
+    "total": {
+      "type": "integer"
+    },
+    "results": {
+      "type": "array",
+      "items": {
+        "$ref": "../../entities/transactions/abstract-transaction.schema.json"
```
Comment (kyranjamie) [inline docs/api/address/get-address-transactions.schema.json:21]: This only returns the base set of abstracts props, not all transactions?

If it returns mempool and confirmed txs, we could write as follows, which would result in nicer compiled type output, 

```json
    "results": {
      "type": "array",
      "items": {
        "type": "object",
        "oneOf": [
          {
            "$ref": "<path to mempooltx>"
            {
          },
            "$ref": "<path to tx>"
          },
        ]
```

File: docs/api/blocks/get-blocks.schema.json [hunk 4]
```diff
@@ -1,18 +1,22 @@
...
     "limit": {
-      "type": "integer"
+      "type": "integer",
+      "maximum": 30,
+      "description": "The number of blocks to return"
     },
     "offset": {
-      "type": "integer"
+      "type": "integer",
+      "description": "The number to blocks to skip (starting at `0`)"
```
Comment (kyranjamie) [inline docs/api/blocks/get-blocks.schema.json:15]:
```suggestion
      "description": "The number to blocks to skip (starting at `0`)",
      "default": 0
```

File: docs/api/address/get-address-balances.schema.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "description": "GET request that returns address balances",
+  "title": "AccountBalance",
+  "type": "object",
+  "required": ["stx", "fungible_tokens", "non_fungible_tokens"],
+  "properties": {
+    "stx": {
+      "type": "object",
```
Comment (kyranjamie) [inline docs/api/address/get-address-balances.schema.json:9]: In a lot of the output types you'll see this prop added:

`[k: string]: unknown | undefined;`

which basically means that type can have any other property added. Often this isn't the case, we can prohibit this where possible by adding this property. Note this is required for nested objects as well.

```suggestion
      "type": "object",
      "additionalProperties": false,
```

File: docs/api/address/get-address-transactions.schema.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "description": "GET request that returns account transactions",
+  "title": "AccountTransactions",
+  "type": "object",
```
Comment (kyranjamie) [inline docs/api/address/get-address-transactions.schema.json:5]:
```suggestion
  "type": "object",
  "additionalProperties": false,
```

File: docs/api/blocks/get-blocks.schema.json [hunk 7]
```diff
@@ -1,18 +1,22 @@
 {
   "$schema": "http://json-schema.org/draft-07/schema#",
-  "description": "GET request that returns transactions",
+  "description": "GET request that returns blocks",
```
Comment (kyranjamie) [inline docs/api/blocks/get-blocks.schema.json:3]:
```suggestion
  "description": "GET request that returns blocks",
  "additionalProperties": false,
```

File: docs/api/core-node/get-account-data.schema.json [hunk 8]
```diff
@@ -0,0 +1,21 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "description": "GET request for account data",
+  "title": "AccountData",
+  "type": "object",
```
Comment (kyranjamie) [inline docs/api/core-node/get-account-data.schema.json:5]:
```suggestion
  "type": "object",
  "additionalProperties": false,
```

File: docs/api/core-node/get-contract-source.schema.json [hunk 9]
```diff
@@ -0,0 +1,18 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "description": "GET request to get contract source",
+  "title": "ContractSource",
+  "type": "object",
```
Comment (kyranjamie) [inline docs/api/core-node/get-contract-source.schema.json:5]:
```suggestion
  "type": "object",
  "additionalProperties": false,
```

File: docs/api/core-node/get-info.schema.json [hunk 10]
```diff
@@ -0,0 +1,61 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "description": "GET request that core node information",
+  "title": "CoreNodeInfo",
+  "type": "object",
```
Comment (kyranjamie) [inline docs/api/core-node/get-info.schema.json:5]:
```suggestion
  "type": "object",
  "additionalProperties": false,
```

File: docs/api/faucet/run-faucet.schema.json [hunk 11]
```diff
@@ -0,0 +1,21 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "description": "POST request that runs the faucet",
+  "title": "RunFaucet",
+  "type": "object",
```
Comment (kyranjamie) [inline docs/api/faucet/run-faucet.schema.json:5]:
```suggestion
  "type": "object",
  "additionalProperties": false,
```

File: docs/api/address/get-address-transactions.schema.json (hunk 3 above)
Comment (agraebe) [reply docs/api/address/get-address-transactions.schema.json]: Great suggestion, thanks!

**Pull request merged**